src/DocumentFormat.OpenXml/Framework/Metadata/ElementMetadata.cs

```diff
@@ -39,10 +43,13 @@ internal class ElementMetadata
             _children = lookup;
         }
 
-        private ElementMetadata()
+        private ElementMetadata(Type type)
         {
+            Type = type;
         }
 
+        public Type Type { get; }
```

Comment:
What's the purpose of this? It's probably fine, but does add some size to the data structure which we create many, many, many of so I'm interested if there's an additional way to do it.

Reply:
The purpose of this is to generate XML documentation in the following form (using the `w:abstractNumId` element and attribute as an example:

```csharp
/// <summary>
/// Represents the w:abstractNumId XML element and attribute.
/// </summary>
/// <remarks>
/// <para>As an XML element, it:</para>
/// <list type="bullet">
/// <item><description>has the following parent XML elements: <see cref="num" />.</description></item>
/// <item><description>has the following XML attributes: <see cref="val" />.</description></item>
/// <item><description>corresponds to the following strongly-typed classes: AbstractNumId.</description></item>
/// </list>
/// <para>As an XML attribute, it:</para>
/// <list type="bullet">
/// <item><description>is contained in the following XML elements: <see cref="abstractNum" />.</description></item>
/// <item><description>corresponds to the following strongly-typed properties: AbstractNum.AbstractNumberId.</description></item>
/// </list>
/// </remarks>
public static readonly XName abstractNumId = w + "abstractNumId";
```

In the above XML documentation, you'll see

- "corresponds to the following strongly-typed classes: **AbstractNumId**" and
- "corresponds to the following strongly-typed properties: **AbstractNum**.AbstractNumberId"

The class names (`AbstractNumId` and `AbstractNum`) are retrieved from the `Type` property. While the corresponding property exists in the `AttributeMetadata` class, it did not exist in the `ElementMetadata` class, so I had to add it.